### DISCUSSION TITLE
Considerations for Channel Consolidation

### DISCUSSION THREAD

**Discussion by josefaidt** [aws-amplify/discord-bot#33] [Ideas]
## **UPDATE January 20, 2022**: modifications and consolidation have been applied 🙂 

**Objective**: to condense and consolidate channels by closing/removing channels rarely visited by community members. Ultimately funneling conversations and increasing visibility; highlighting the community intent.

Using the table below I've made some notes on my thoughts for consolidating channels based on usage statistics.

| Existing Category    | Existing Channel             | New Category | New Channel      | Remove? | Remove Reason                         |
| -------------------- | ---------------------------- | ------------ | ---------------- | ------- | ------------------------------------- |
| AWS Amplify          | announcements                | ~community~ none    | (add emoji?)     | -       |                                       |
| AWS Amplify          | rules                        | ~onboarding~ none   | (add emoji?)     | -       |                                       |
| AWS Amplify          | lobby                        | community    | general          | -       |                                       |
| AWS Amplify          | introductions                | ~onboarding~ community   | 👋 introductions | -       |                                       |
| Contribution         | contributing                 | community    |                  | -       |                                       |
| Contribution         | contribute-to-android        | -            | -                | yes     | roll into single contributing channel |
| Contribution         | contribute-to-docs           | -            | -                | yes     | roll into single contributing channel |
| Contribution         | contribute-to-flutter        | -            | -                | yes     | roll into single contributing channel |
| Contribution         | contribute-to-ios            | -            | -                | yes     | roll into single contributing channel |
| Contribution         | contribute-to-javascript     | -            | -                | yes     | roll into single contributing channel |
| Community            | amplify-with-friends         | -            | -                | yes     | close in favor of events feature      |
| Community            | contributors (private)       | -            | -                | -       | ?                                     |
| Community            | community-builders (private) | -            | -                | -       | ?                                     |
| Community            | discord-feedback             | -            | -                | -       |                                       |
| Community            | first-time-contributor       | -            | -                | yes     | roll into single contributing channel |
| Community            | events                       | -            | -                | yes     | close in favor of events feature      |
| Community            | job-postings                 | -            | -                | -       |                                       |
| Community            | random                       | -            | -                | yes     | close in favor of #general            |
| Community            | resources                    | -            | -                | ?       | merge resources and showcase?         |
| Community            | showcase                     | -            | -                | ?       | merge resources and showcase?         |
| Amplify Office Hours | office-hours                 | community    | -                | -       |                                       |
| Amplify Office Hours | Amplify Office Hours (stage) | community    | -                | -       |                                       |

If we are to set up additional announcement-type channels (i.e. places to promote GitHub releases), we should consider an announcement-specific category

This is a rough draft and there are a few ?'s to fill in and get feedback on, and we would greatly appreciate additional feedback. 
A few questions/notes:

- how do we feel about an `onboarding` category vs no category at all for the proposed channel migrations
- for channels we add emojis to we should expect a lack in ability to use the cmd+k switcher menu to navigate to these channels (example: for `👋 introductions` it is only searchable by searching for `👋 ...` instead of just `introductions`). Emojis should be limited to low-impact channels
- as part of the consolidation we should consider moving the community section above the help channels

**Comment by billygerhard** [comment]
What if you add the emoji to the end? That should fix searching issue.

**Reply by josefaidt** [reply]
You are absolutely right! I initially based those "emoji-fied" names off of [CloudFlare's developer server](https://discord.gg/b2kKfenB)
![image](https://user-images.githubusercontent.com/5033303/145488394-b56857d6-4219-4cef-b9f5-3bd6ef4cc9d9.png)

![image](https://user-images.githubusercontent.com/5033303/145488307-ea86aea7-167d-40f7-8be0-1a0d97c6e87b.png)
